File: AsciiProgressBar.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
double P;
cin>>P;

int percentage = floor(P);
int progress = percentage/10;

cout<<"[";
for(int i=0; i<progress; i++) cout<<"+";
for(int i=progress; i<10; i++) cout<<".";
cout<<"] "<<percentage<<"%"<<endl;

    return 0;
}

File: AsciiProgressBar.swift
import Foundation

if let input = readLine(), let P = Double(input){
    let percentage = Int(floor(P))
    let progress = percentage/10
    var bar = "["
    for _ in 0..<progress{
        bar += "+"
    }
    for _ in progress..<10{
        bar += "."
    }
    bar += "] "

    print("\(bar)\(percentage)%")
}